client/payments/data/reducer.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Internal dependencies
+ */
+import TYPES from './action-types';
+
+const DEFAULT_STATE = {
+	paymentGateways: [],
+};
+
+const reducer = ( state = DEFAULT_STATE, action ) => {
+	switch ( action.type ) {
+		case TYPES.RECEIVE_PAYMENT_GATEWAYS:
+			return {
+				...state,
+				paymentGateways: action.paymentGateways,
+			};
+		case TYPES.RECEIVE_PAYMENT_GATEWAY:
+			const paymentGateways = state.paymentGateways.filter(
```

Comment:
Does it matter to you that this doesn't retain the order of the array when there is a replacement? It might be ok for your use case, but the next person that uses this data store may find the lack of maintained order in the array surprising.

I'm also not sure if this replacement is something you expect in normal operation or if its an edge case, but assuming its normal, you could check if the item is in the array first.

e.g.

```
const index = state.paymentGateways.findIndex((gateway) => gateway.id === action.paymentGateway.id);

if (index === -1) {
  return {
     ...state,
     paymentGateways: [...state.paymentGateways, action.paymentGateway]
   }
}

return {
  ...state,
  paymentGateways: [...array.slice(0, index), action.paymentGateway, ...array.slice(index + 1)]
}
```
I haven't tested this code, just a rough idea 😄

Reply:
That's a good catch 👍 . Now that I think about it..I think the order matters when you use the array to render the payments. I think I should find the index and then replace it.